src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/history/NetworkHistory.java

```diff
@@ -13,72 +13,95 @@
  *   limitations under the License.
  */
 
-package org.openkilda.wfm.topology.network.storm.spout;
+package org.openkilda.wfm.topology.network.storm.bolt.history;
 
+import static org.openkilda.wfm.share.zk.ZooKeeperSpout.FIELD_ID_LIFECYCLE_EVENT;
+
+import org.openkilda.bluegreen.LifecycleEvent;
+import org.openkilda.bluegreen.Signal;
 import org.openkilda.model.SwitchId;
 import org.openkilda.persistence.PersistenceManager;
-import org.openkilda.persistence.context.PersistenceContextRequired;
 import org.openkilda.wfm.AbstractBolt;
 import org.openkilda.wfm.CommandContext;
+import org.openkilda.wfm.share.zk.ZkStreams;
+import org.openkilda.wfm.share.zk.ZooKeeperBolt;
 import org.openkilda.wfm.topology.network.model.facts.HistoryFacts;
 import org.openkilda.wfm.topology.network.service.ISwitchPrepopulateCarrier;
 import org.openkilda.wfm.topology.network.service.NetworkHistoryService;
 import org.openkilda.wfm.topology.network.storm.ComponentId;
 import org.openkilda.wfm.topology.network.storm.bolt.speaker.SpeakerRouter;
 import org.openkilda.wfm.topology.network.storm.bolt.sw.command.SwitchHistoryCommand;
 
-import org.apache.storm.spout.SpoutOutputCollector;
-import org.apache.storm.task.TopologyContext;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.storm.topology.OutputFieldsDeclarer;
-import org.apache.storm.topology.base.BaseRichSpout;
 import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
 import org.apache.storm.tuple.Values;
 
-import java.util.Map;
-
-public class NetworkHistory extends BaseRichSpout implements ISwitchPrepopulateCarrier {
-    public static final String SPOUT_ID = ComponentId.NETWORK_HISTORY.toString();
+@Slf4j
+public class NetworkHistory extends AbstractBolt implements ISwitchPrepopulateCarrier {
```

Comment:
It should have the suffix `Handler` because it is bolt now.

Reply:
Suffix has been added